File: exercism/javascript/series/series.js
const fp = require('lodash/fp')

const zip = fp.zip.convert({fixed: false})

class Series {
  constructor (n) {
    this.digits = n.split('').map(fp.parseInt(10))
  }
  slices (n) {
    if (n > this.digits.length) throw new Error('Slice size is too big.')
    if (n === 1) return this.digits.map(d => [d])
    const steps = fp.map(fp.drop(fp, this.digits), fp.range(0, n))
    return zip.apply(null, steps).slice(0, this.digits.length + 1 - n)
  }
}

module.exports = Series

File: exercism/cpp/series/series.cpp
#include "series.h"
#include <iostream>
#include <string>
#include <vector>

std::vector<int> series::digits(std::string chars) {
  std::vector<int> digits;
  for (auto &ch : chars) {
    int d = ch - '0';
    if (d < 0 || d > 9)
      throw std::domain_error("Invalid char");
    digits.push_back(d);
  }
  return digits;
}

std::vector<std::vector<int>> series::slice(std::string chars, int n) {
  std::vector<int> digits = series::digits(chars);
  int size = digits.size();

  if (size < n)
    throw std::domain_error("Insufficient digits to slice");

  int num_slices = size - n + 1;
  std::vector<std::vector<int>> slices(num_slices, std::vector<int>(n));

  for (int i = 0; i < num_slices; i++) {
    for (int j = 0; j < n; j++) {
      slices[i][j] = digits.at(i + j);
    }
  }
  return slices;
}
